core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -805,10 +805,22 @@ void setNonce(long nonce) {
         this.hash = null;
     }
 
-    /** Returns an unmodifiable list of transactions held in this block, or null if this object represents just a header. */
+    /**
+     * Returns an unmodifiable list of transactions held in this block. This can only be called on regular
+     * (non-header-only) blocks.
+     *
+     * @return transactions in this block, can be empty
+     */
+    public List<Transaction> transactions() {
+        checkState(!isHeaderOnly(), () -> "block is header-only");
+        return Collections.unmodifiableList(transactions);
```

Comment:
Just a reminder that the underlying list is technically still modifiable as are the transaction contained within.